src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/service/DescriptionProviderService.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.companieshouse.certifiedcopies.orders.api.service;
+
+import org.apache.commons.lang.text.StrSubstitutor;
+import org.springframework.stereotype.Service;
+import org.yaml.snakeyaml.Yaml;
+import uk.gov.companieshouse.certifiedcopies.orders.api.logging.LoggingUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+
+@Service
+public class DescriptionProviderService {
+
+    private static final String COMPANY_NUMBER_KEY = "company_number";
+    private static final String CERTIFIED_COPY_DESCRIPTION_KEY = "certified-copy-description";
+    private static final String COMPANY_CERTIFIED_COPY_DESCRIPTION_KEY = "company-certified-copy";
+
+    private static final String ORDERS_DESCRIPTIONS_FILEPATH = "api-enumerations/orders_descriptions.yaml";
+
+    private final String companyCertifiedCopyDescription;
+
+    private static final String LOG_MESSAGE_FILE_KEY = "file";
+
+    public DescriptionProviderService() {
+        final File ordersDescriptionsFile = new File(ORDERS_DESCRIPTIONS_FILEPATH);
+        companyCertifiedCopyDescription = getCompanyCertifiedCopyDescription(ordersDescriptionsFile);
+    }
+
+    public DescriptionProviderService(final File ordersDescriptionsFile) {
+        companyCertifiedCopyDescription = getCompanyCertifiedCopyDescription(ordersDescriptionsFile);
+    }
+
+    /**
+     * Gets the configured description.
+     * @param companyNumber the company number making up part of the description
+     * @return the configured description, or <code>null</code> if none found.
+     */
+    public String getDescription(final String companyNumber) {
+        if (companyCertifiedCopyDescription == null) {
+            // Error logged again here at time description is requested.
+            Map<String, Object> logMap = new HashMap<>();
+            logMap.put(COMPANY_CERTIFIED_COPY_DESCRIPTION_KEY, COMPANY_CERTIFIED_COPY_DESCRIPTION_KEY);
```

Comment:
Should the key and value be the same?

Reply:
Certificates has the same thing. But it seems unnecessary to put it in. I think the error below is sufficient. So I'll removed the logmap.

Reply:
I was thinking that perhaps the key might be a name that describes what the value actually represents, rather than the value itself?

Reply:
Oh I see, I'll change it to that.